lxd/resource_lxd_container.go

```diff
@@ -96,38 +98,40 @@ func resourceLxdContainerCreate(d *schema.ResourceData, meta interface{}) error
 
 	//client.Init = (name string, imgremote string, image string, profiles *[]string, config map[string]string, ephem bool)
 	var resp *lxd.Response
-	if resp, err = client.Init(name, remote, d.Get("image").(string), &profiles, nil, nil, ephem); err != nil {
+	if resp, err = client.Init(name, remote, d.Get("image").(string), &profiles, config, nil, ephem); err != nil {
 		return err
 	}
 
-	d.SetId(name)
-
 	err = client.WaitForSuccess(resp.Operation)
-	if err == nil {
-		// Start container
-		resp, err = client.Action(name, shared.Start, -1, false, false)
-		if err != nil {
-			// Container has been created, but daemon rejected start request
-			return err
-		}
-
-		if err := client.WaitForSuccess(resp.Operation); err != nil {
-			// Container could not be started
-			return err
-		}
+	if err != nil {
+		return err
+	}
 
-		return resourceLxdContainerRead(d, meta)
+	// Start container
+	resp, err = client.Action(name, shared.Start, -1, false, false)
+	if err != nil {
+		// Container has been created, but daemon rejected start request
+		return err
 	}
 
-	// Resource didn't create
-	d.SetId("")
+	if err := client.WaitForSuccess(resp.Operation); err != nil {
+		// Container could not be started
+		return err
+	}
 
-	return err
+	d.SetId(name)
+	return resourceLxdContainerRead(d, meta)
```

Comment:
I probably should have rolled this into a separate PR. The refactor of this flow more closely aligns with other core Terraform providers. It's generally best to wait until the very end to set the ID. If there was an error, it would have been triggered somewhere above and this point should have never been hit.